libcxx/src/atomic.cpp

```diff
@@ -69,17 +69,17 @@ extern "C" int __ulock_wait(
     uint32_t operation, void* addr, uint64_t value, uint32_t timeout); /* timeout is specified in microseconds */
 extern "C" int __ulock_wake(uint32_t operation, void* addr, uint64_t wake_value);
 
-#  define UL_COMPARE_AND_WAIT 1
+#  define UL_COMPARE_AND_WAIT64 5
```

Comment:
Could you explain the magic number?

Reply:
yeah it is a bit fiddly. we (with Louis) couldn't even find the header in the SDK. it is some sort of "internal" API
https://github.com/apple/darwin-xnu/blob/2ff845c2e033bd0ff64b5b6aa6063a1f8f65aa32/bsd/sys/ulock.h#L82

Reply:
Thank you.

Could you add that link as a comment in the source for the next reader who has to go on the hunt for it?

Reply:
Good idea. thanks